src/BTCPayServer.Lightning.Common/ILightningClient.cs

```diff
@@ -9,17 +7,18 @@ namespace BTCPayServer.Lightning
 {
     public interface ILightningClient
     {
-        Task<LightningInvoice> GetInvoice(string invoiceId, CancellationToken cancellation = default(CancellationToken));
-        Task<LightningInvoice> CreateInvoice(LightMoney amount, string description, TimeSpan expiry, CancellationToken cancellation = default(CancellationToken));
-        Task<LightningInvoice> CreateInvoice(CreateInvoiceParams createInvoiceRequest, CancellationToken cancellation = default(CancellationToken));
-        Task<ILightningInvoiceListener> Listen(CancellationToken cancellation = default(CancellationToken));
-        Task<LightningNodeInformation> GetInfo(CancellationToken cancellation = default(CancellationToken));
-        Task<PayResponse> Pay(string bolt11, CancellationToken cancellation = default(CancellationToken));
-        Task<OpenChannelResponse> OpenChannel(OpenChannelRequest openChannelRequest, CancellationToken cancellation = default(CancellationToken));
+        Task<LightningInvoice> GetInvoice(string invoiceId, CancellationToken cancellation = default);
+        Task<LightningInvoice> CreateInvoice(LightMoney amount, string description, TimeSpan expiry, CancellationToken cancellation = default);
+        Task<LightningInvoice> CreateInvoice(CreateInvoiceParams createInvoiceRequest, CancellationToken cancellation = default);
+        Task<ILightningInvoiceListener> Listen(CancellationToken cancellation = default);
+        Task<LightningNodeInformation> GetInfo(CancellationToken cancellation = default);
+        Task<PayResponse> Pay(string bolt11, float? maxFeePercent, CancellationToken cancellation = default);
```

Comment:
I was thinking about this too, but refrained from it, because there are not many fields across the implementations. But yeah, you are right in terms of making it more future proof. I'll update it.

Reply:
Done in [45a096c](https://github.com/btcpayserver/BTCPayServer.Lightning/pull/61/commits/45a096ce71790211b996b911db13e19e815312c2).